### PR TITLE
Implement %%, %%=, and operator""

### PR DESCRIPTION
**Pull request by LemonInTheDark**
The float stuff was easy. Operator was too, just took a bit to realize how it worked

Nothin much else to say, outside that I needed to roll my own operation for this one, since float remainder is kinda hard to find (and I couldn't after cursory reading)

### CODE REVIEW

File: crates/dreammaker/src/parser.rs [hunk 1]
```diff
@@ -944,6 +950,8 @@ impl<'ctx, 'an, 'inp> Parser<'ctx, 'an, 'inp> {
             } else {
                 last_part.push_str("[]");
             }
+        } else if self.exact(Punct(EmptyQuote))?.is_some() {
```
Comment (SpaceManiac) [inline crates/dreammaker/src/parser.rs:953]: A less shocking change might be to look for an empty `Token::String` here.

File: crates/dreammaker/src/lexer.rs [hunk 2]
```diff
@@ -71,9 +71,12 @@ table! {
     "!",   Not;
     "!=",  NotEq;
     "\"",  DoubleQuote;
+    "\"\"",  EmptyQuote;
```
Comment (SpaceManiac) [inline crates/dreammaker/src/lexer.rs:74]: Now that this has its own token, the parser no longer recognizes it.

**Review by SpaceManiac — CHANGES_REQUESTED** [review]

File: crates/dreammaker/src/lexer.rs (hunk 2 above)
Comment (LemonInTheDark) [reply crates/dreammaker/src/lexer.rs:74]: Oh, I see. I was unsure how the token system actually worked. 
I'll see about using Token::string instead, cause that's bad you're right

File: crates/dreammaker/src/parser.rs (hunk 1 above)
Comment (LemonInTheDark) [reply crates/dreammaker/src/parser.rs]: handled, I think?

**Pull request merged**